src/dataframe/dataframe.jl

```diff
@@ -791,19 +791,21 @@ function Base.delete!(df::DataFrame, inds)
         throw(BoundsError(df, (inds, :)))
     end
     # we require ind to be stored and unique like in Base
+    # otherwise an error will be thrown and the data frame will get corrupted
     foreach(col -> deleteat!(col, inds), _columns(df))
-    df
+    return df
 end
 
 function Base.delete!(df::DataFrame, inds::AbstractVector{Bool})
     if length(inds) != size(df, 1)
         throw(BoundsError(df, (inds, :)))
     end
-    drop = findall(inds)
```

Comment:
I think this was intentional: computing integer indices once is faster than using logical indexing for each column, as it requires computing the sum again and again. Though a comment wouldn't hurt.

Reply:
indeed - now I remember thinking about it some tome ago. I will revert. Anyway this PR should not be merged before #2199 as it is conflicting. Thank you!